modules/raft/src/main/java/org/apache/ignite/internal/raft/storage/impl/VolatileLogStorageFactory.java

```diff
@@ -34,7 +83,21 @@ public void start() {
     /** {@inheritDoc} */
     @Override
     public LogStorage createLogStorage(String uri, RaftOptions raftOptions) {
-        return new LocalLogStorage(raftOptions);
+        return new VolatileLogStorage(createLogStorageBudget());
+    }
+
+    private LogStorageBudget createLogStorageBudget() {
+        return newBudget(logStorageBudgetConfig);
+    }
+
+    private LogStorageBudget newBudget(LogStorageBudgetView logStorageBudgetConfig) {
+        LogStorageBudgetFactory factory = budgetFactories.get(logStorageBudgetConfig.name());
+
+        if (factory == null) {
+            throw new IgniteInternalException("Cannot find a log storage budget by name '" + logStorageBudgetConfig.name() + "'");
```

Comment:
Is there any possibility to not use deprecated constructor for` IgniteInternalException`? 
Maybe we should introduce some error codes for that scenario

Reply:
Added a TODO to address this later